bee-network/src/protocols/gossip.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright 2020 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::NETWORK_ID;
+
+use futures::prelude::*;
+use libp2p::core::{
+    muxing::{StreamMuxerBox, SubstreamRef},
+    InboundUpgrade, Negotiated, OutboundUpgrade, UpgradeInfo,
+};
+
+use std::{
+    iter,
+    sync::{atomic::Ordering, Arc},
+};
+
+pub type GossipSubstream = Negotiated<SubstreamRef<Arc<StreamMuxerBox>>>;
+
+#[derive(Default, Debug, Clone)]
+pub struct GossipProtocol;
+
+impl UpgradeInfo for GossipProtocol {
+    type Info = Vec<u8>;
+    type InfoIter = iter::Once<Self::Info>;
+
+    fn protocol_info(&self) -> Self::InfoIter {
+        iter::once(
+            format!("/iota-gossip/{}/1.0.0", NETWORK_ID.load(Ordering::Relaxed))
```

Comment:
Global state :\

Reply:
yea, you mean the `NETWORK_ID`? That's not really an issue, because it's only set once during initialization. We could think about using a different primitive, that is more appropriate in such a scenario, but the alternative would be to have bigger signatures as this value would have to be passed in a chain, which increases complexity and decreases readibility. Or is your concern headed towards another angle?